src/components/TrapFocus/TrapFocus.tsx

```diff
@@ -68,25 +78,25 @@ export class TrapFocus extends React.PureComponent<TrapFocusProps, State> {
   };
 
   private handleBlur = (event: FocusEvent) => {
-    const {relatedTarget} = event;
+    const {relatedTarget: currentTarget} = event;
     const {focusTrapWrapper} = this;
     const {trapping = true} = this.props;
 
-    if (relatedTarget == null || trapping === false) {
+    if (trapping === false) {
       return;
     }
 
     if (
       focusTrapWrapper &&
-      !focusTrapWrapper.contains(relatedTarget as HTMLElement) &&
-      !closest(relatedTarget as HTMLElement, '[data-polaris-overlay]')
```

Comment:
@AndrewMusgrave Do you think we need this line? I can't think of a situation when we would need to find an element with the property `data-polaris-overlay`?

Reply:
Hard to tell because I couldn't find the original commit. But if it's not part of the issue I would leave it as to not introduce more issues. It is extra precaution.

Reply:
I'm not sure tbh, `d-p-overlay` is legacy but were not 100% confident in removing it. As Dan mentioned it removing it required for this change?

Reply:
Removing it is not required for this change, though I was mostly hoping we could remove it if it was not being used as things appear to work fine without it. But we can keep it if we're not 100% confident in removing it.